webapi/src/main/scala/org/knora/webapi/slice/admin/domain/model/MaintenanceAction.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2021 - 2024 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.admin.domain.model
+
+import zio.Task
+import zio.ZIO
+
+import org.knora.webapi.store.triplestore.api.TriplestoreService
+
+trait MaintenanceAction[A] {
+  val triplestoreService: TriplestoreService
+
+  /**
+   * Check if the action should be executed.
+   *
+   * This method can be implemented to define a condition under which the action should be executed or not.
+   * This mechanism may be used to prevent the execution of an action if it is not necessary,
+   * if this would be a more expensive operation.
+   * This would typically be an ASK query to the triplestore.
+   *
+   * @return true if the action should be executed, false otherwise
+   */
+  def shouldExecute: Task[Boolean] = ZIO.succeed(true)
+
+  /**
+   * Executes the maintenance action.
+   *
+   * @param params optional parameters as provided by the caller
+   */
+  def execute(params: Option[A] = None): Task[Unit]
```

Comment:
If we would type the `params` as `A` you could use `Any` for `A` to indicate that no params are necessary. This saves us from double checking in the `execute` method if params is  a `Some`.

Reply:
done in [a26c378](https://github.com/dasch-swiss/dsp-api/pull/3206/commits/a26c3788f270efb086898d023a71dc82b678d96b), thanks for the hint, looks much nicer.

We can then type it as `Any` or `Nothing`, both seem to work.  
The only downside is that I cannot provide a default argument anymore, so instead of calling `FooAction(null).execute()`, I have to do `FooAction.execute(())`. That's slightly less intuitive, but I think it's ok.  
One option to get this to work might be to have a trait `NoParamMaintenanceAction` that constraints the type... but I don't think it's worth it.